openedx/features/badging/models.py

```diff
@@ -18,3 +23,79 @@ class Badge(models.Model):
 
     def __unicode__(self):
         return self.name
+
+    @classmethod
+    def get_remaining_badges(cls, user_id, community_id):
+        course_id = get_course_id_by_community_id(community_id)
+
+        if course_id is not CourseKeyField.Empty:
+            badge_type = CONVERSATIONALIST[0]
+            badges_in_community = UserBadge.objects.filter(user_id=user_id, community_id=community_id) \
+                                                   .exclude(course_id=CourseKeyField.Empty)
+        else:
```

Comment:
lets not rely on else and the user provided data, lets also check  if the community is a team community. (specially because this is a get method)
Also we can use this query:
```
badges_in_community = UserBadge.objects.filter(
            user_id=user_id,
            community_id=community_id,
            badge__type=badge_type
        )
```

Reply:
Badge type sent through template in a data field and sent with request. Suggested query used.